src/Hosting/Hosting/test/HostingApplicationTests.cs

```diff
@@ -87,7 +88,97 @@ public void CreateContextCreatesNewContextIfNotUsingDefaultHttpContextFactory()
+
+            var activityFeature = context.HttpContext.Features.Get<IHttpActivityFeature>();
+            Assert.NotNull(activityFeature);
+            Assert.NotNull(activityFeature.Activity);
+            Assert.Equal(HostingApplicationDiagnostics.ActivityName, activityFeature.Activity.DisplayName);
+            var initialActivity = Activity.Current;
+
+            // Create nested dummy Activity
+            using var _ = dummySource.StartActivity("DummyActivity");
+
+            Assert.Same(initialActivity, activityFeature.Activity);
+            Assert.NotEqual(Activity.Current, activityFeature.Activity);
+
+            // Act/Assert
+            hostingApplication.DisposeContext(context, null);
+        }
+
+        private class TestHttpActivityFeature : IHttpActivityFeature
+        {
+            public Activity Activity { get; set; }
+        }
+
+        [Fact]
+        public void IHttpActivityFeatureIsAssignedToIfItExists()
+        {
+            var testSource = new ActivitySource(Path.GetRandomFileName());
+            var dummySource = new ActivitySource(Path.GetRandomFileName());
+            using var listener = new ActivityListener
+            {
+                ShouldListenTo = activitySource => (ReferenceEquals(activitySource, testSource) ||
+                                                    ReferenceEquals(activitySource, dummySource)),
+                Sample = (ref ActivityCreationOptions<ActivityContext> _) => ActivitySamplingResult.AllData
+            };
+            ActivitySource.AddActivityListener(listener);
+
+            var hostingApplication = CreateApplication(activitySource: testSource);
+            var httpContext = new DefaultHttpContext();
+            httpContext.Features.Set<IHttpActivityFeature>(new TestHttpActivityFeature());
+            var context = hostingApplication.CreateContext(httpContext.Features);
+
+            var activityFeature = context.HttpContext.Features.Get<IHttpActivityFeature>();
+            Assert.NotNull(activityFeature);
+            Assert.IsType<TestHttpActivityFeature>(activityFeature);
+            Assert.NotNull(activityFeature.Activity);
+            Assert.Equal(HostingApplicationDiagnostics.ActivityName, activityFeature.Activity.DisplayName);
+            var initialActivity = Activity.Current;
+
+            // Create nested dummy Activity
+            using var _ = dummySource.StartActivity("DummyActivity");
+
+            Assert.Same(initialActivity, activityFeature.Activity);
+            Assert.NotEqual(Activity.Current, activityFeature.Activity);
+
+            // Act/Assert
+            hostingApplication.DisposeContext(context, null);
+        }
+
+        [Fact]
+        public void IHttpActivityFeatureIsNotPopulatedWithoutAListener()
+        {
+            var hostingApplication = CreateApplication();
+            var httpContext = new DefaultHttpContext();
+            httpContext.Features.Set<IHttpActivityFeature>(new TestHttpActivityFeature());
```

Comment:
```suggestion
```

IMO test the default activity feature is there when there is no activity.

Reply:
Well it isn't because there is no server here to set it on the context